test/ilpQuoterSpec.js

```diff
@@ -0,0 +1,110 @@
+'use strict'
+
+const nock = require('nock')
+nock.enableNetConnect(['localhost'])
+const expect = require('chai').expect
+
+const NoAmountSpecifiedError = require('../src/errors/no-amount-specified-error')
+const AssetsNotTradedError = require('../src/errors/assets-not-traded-error')
+const ServerError = require('five-bells-shared/errors/server-error')
+const Backend = require('../src/backends/ilp-quoter')
+
+describe('ILPQuoter', function () {
+  beforeEach(function * () {
+    this.BACKEND = 'ilp-quote'
+    this.BACKEND_URI = 'http://marketmaker.quoter.com'
+
+    process.env.CONNECTOR_PAIRS =
+      '[["USD@https://localhost:3000","EUR@https://localhost:4001"],' +
+      '["EUR@https://localhost:4001","USD@https://localhost:3000"],' +
+      '["USD@https://localhost:4000","EUR@https://localhost:3001"],' +
+      '["EUR@https://localhost:3001","USD@https://localhost:4000"]]'
+    process.env.CONNECTOR_BACKEND = this.BACKEND
+    process.env.CONNECTOR_BACKEND_URI = this.BACKEND_URI
+    const loadConfig = require('../src/lib/config')
+    this.backend = new Backend(loadConfig(), {})
+  })
+
+  afterEach(function () {
+    nock.cleanAll()
+  })
+
+  function * yieldAndAssertException (action, exception) {
+    let err
+    try {
+      yield action
+    } catch (e) {
+      err = e
+    }
+    expect(err).to.be.an.instanceOf(exception)
+  }
+
+  describe('quoter flow', function () {
+    it('should make sure the backend PUT /pair is called', function * () {
+      const scope = nock(this.BACKEND_URI)
+                      .put('/pair/EUR/USD').reply(200)
+                      .put('/pair/USD/EUR').reply(200)
+      yield this.backend.connect()
+      scope.isDone()
+    })
+
+    it('should fail for a quote with a missing source_ledger', function * () {
+      const quote = { source_amount: '123', destination_ledger: 'https://localhost:4000' }
+      yield yieldAndAssertException(this.backend.getQuote(quote), AssetsNotTradedError)
```

Comment:
I think the issue is that co-mocha doesn't let you use that `expect` with generator functions.

I mentioned using `co` because mocha works just fine with promises.

```
co.wrap(this.backend.getQuote)(quote).catch((err) => expect(err)....
```